### PR TITLE
Fix mapboxgl symbollayer style for v1

### PR DESCRIPTION
**Pull request by OttyLab**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests). --> ran `npm test mapbox-gl/v1` that is added in this PR

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://docs.mapbox.com/mapbox-gl-js/style-spec/layers/#layout-symbol-text-justify, https://docs.mapbox.com/mapbox-gl-js/style-spec/layers/#layout-symbol-icon-ignore-placement>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

## Additional information
This PR is as same fix as #52616. #52616 was for v2 mapbox-gl and this PR is for v1.13.x. To support v1, I created v1 folder and copied files from 51bfb1191e that is the latest for v1.13.x.

### CODE REVIEW

File: types/mapbox-gl/v1/tslint.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        "no-unnecessary-generics": false,
+        "no-unnecessary-qualifier": false,
+        "no-var-keyword": false,
+        "object-literal-shorthand": false,
+        "only-arrow-functions": false,
+        "prefer-const": false,
+        "space-before-function-paren": false,
+        "strict-export-declare-modifiers": false,
+        "triple-equals": false
+    }
```
Comment (elibarzilay) [inline types/mapbox-gl/v1/tslint.json:23]: Stick to the required tslint form

Comment (OttyLab) [reply types/mapbox-gl/v1/tslint.json:23]: @elibarzilay 

Thank you for the review! Does "required tslint form" mean below in README?

> The linter configuration file, tslint.json should contain { "extends": "dtslint/dt.json" }, and no additional rules.

If the rules are disabled, so many errors occurs. 

The purpose of this PR is just to want to fix a minor issue that affects mapbox GL JS v1.13.x function (it's a maintenance phase version in Mapbox) . Therefore, I want to keep almost all as-is.

**Comment by andrewbranch** [comment]
It looks like mapbox-gl v1 is still getting a lot of downloads, so I’m ok with this. Would like to get an owner approval though.

**Review by dmytro-gokun — APPROVED** [review]

**Review by dmytro-gokun — APPROVED** [review]

**Comment by dmytro-gokun** [comment]
Ready to merge

**Pull request merged**